### ISSUE TITLE
Fix or remove the post-election statistics

### ISSUE THREAD

**Issue by BjarniRunar** [piratar/wasa2il#155]
Recording this here because I think it was at risk of getting forgotten.

After the last primaries, we displayed a bunch of statistics about how people voted and allowed people to download reports that were even more detailed. This was done to satisfy people's (and the media's) curiosity, and in the name of transparency.

In my opinion, I made a mistake here (see PR #144) - the statistics were too detailed. This caused at least the following problems:

1. People who got very few votes (or were placed near the bottom) were upset / hurt / humiliated
2. Candidates were able to verify whether their friends had voted "as promised"

Number one is simply unnecessary. There is no need to publish detailed statistics about people who "lost", so at the very least we should be much more aggressive about truncating the statistics and only publishing details about "winners".

Number two is more serious and also harder to fix; the whole point of anonymous ballots is to prevent peer pressure of any kind from affecting the elections - in the extreme such peer pressure can be blackmail, bribery or coercion, but it can also just be friends asking friends for favours. I personally heard candidates say things along the lines of: "ten people promised to put me in first place, but I only placed first 8 times, hmmmmmmmm..." Preventing this problem is difficult without disabling the statistics entirely, because it implies that we cannot safely publish statistics unless the number of cast ballots per candidate is relatively high - if our future primaries are anything like last summer, that is actually quite unlikely.

I am not sure how to fix this; we definitely need to publish less detailed stats, but how much is safe depends a great deal on each individual election (how many voters, how many candidates) and it may be hard to strike a good balance. This does need to be fixed before we run another election though, and the bare minimum fix is to disable the stats - they aren't safe to use at the moment.

This issue also relates to #80.

**Comment by stjanigunnars** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Fix or remove the post-election statistics](https://trello.com/c/tzE2Kd7R/3-fix-or-remove-the-post-election-statistics)

**Comment by helgihg** [comment]
I don't know how to proceed with this, unless it it's by disabling the statistics feature altogether.

**Comment by BjarniRunar** [comment]
@helgihg That is the safe thing to do, I also am unsure how to proceed. Unfortunately I think people liked the pretty graphs and will be disappointed...

Although flawed, these stats are still *way* better than publishing raw ballot data as was originally planned. There is a certain amount of pressure on us to be "transparent" and publish some stats to contextualize the results.

Maybe it would make sense to sit down in a small group of election-geeks and discuss the threat models and debate what is "anonymous enough"? The reason for doing this offline would be to not leave online a convenient how-to explaining exactly how to mess with us... I know @smari would have opinions, possibly @bjornlevi as well?

**Comment by helgihg** [comment]
Right. I will disable it, at least. I don't know these methods well enough, and I need a way to move this thing forward. So I'll disable it for now, and if the election-method geeks come up with a method of showing data in a responsible way, then we can change the functionality.

**Comment by helgihg** [comment]
So this is how I solved it. I made three options, which can be combined in any way that the admin of each election sees fit:

(A): Publish basic ballot statistics

(B): Publish ballot statistics for each candidate

(C): Publish advanced statistics (downloadable)

Any combination of the aforementioned can be combined as the admin sees fit at any given election. Hot potato gently tossed over, into the hands of whoever makes decisions on any given election.
